### PR TITLE
feat: adds OS filesystem.

### PR DESCRIPTION
**Pull request by jcchavezs**
OS filesystem is handy when combining CRS and local rules.

### CODE REVIEW

File: io/os.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// Copied from https://github.com/corazawaf/coraza/blob/v3/dev/internal/io/file.go
+
+// OSFS implements fs.FS using methods on os to read from the system.
+// Note that this implementation is not a compliant fs.FS, as they should only
+// accept posix-style, relative paths, but as this is an internal implementation
```
Comment (anuraaga) [inline io/os.go:13]: If this is exposed it's not an internal implementation detail anymore and a public, non-compliant implementation of `fs.FS`. I don't know if it's really a big deal in practice but it is something to think about

Comment (jcchavezs) [reply io/os.go:13]: I would say as long as it works for coraza it is OK to have it.

Comment (anuraaga) [reply io/os.go:13]: Sounds ok. Let's fix up the comment to read like a public API

**Review by M4tteoP — APPROVED** [review]

**Comment by jcchavezs** [comment]
Merging as it is now, we can address post merge commits.

**Pull request merged**